pycbc/events/status.py

```diff
@@ -0,0 +1,252 @@
+    """Return the times where the flag is active 
+
+    Parameters
+    ----------
+    ifo: string
+        The interferometer to query (H1, L1).
+    segment_name: string
+        The status flag to query from LOSC.
+    start_time: int 
+        The starting gps time to begin querying from LOSC
+    end_time: int 
+        The end gps time of the query
+    source: str, Optional
+        Choice between "GWOSC" or "dqsegdb". If dqsegdb, the server option may
+        also be given. The default is to try GWOSC first then try dqsegdb.
+    server: str, Optional
+        The server path. Only used with dqsegdb atm.
+    veto_definer: str, Optional
+        The path to a veto definer to define groups of flags which
+        themselves define a set of segments.
+
+    Returns
+    ---------
+    segments: glue.segments.segmentlist
+        List of segments
+    """
+    if source == 'GWOSC' or source == 'any':
+        ### Special cases as the LOSC convention is backwards from normal
+        ### LIGO / Virgo operation!!!!
+        if (('_HW_INJ' in segment_name and not 'NO' in segment_name) or 
+           'VETO' in segment_name):
+            data = query_flag(ifo, 'DATA', start_time, end_time)
+             
+            if '_HW_INJ' in segment_name:
+                name = 'NO_' + segment_name    
+            else:
+                name = segment_name.replace('_VETO', '')
+
+            negate = query_flag(ifo, name, start_time, end_time)
+            return (data - negate).coalesce()
+
+        duration = end_time - start_time
+        url = 'https://www.gw-openscience.org/timeline/segments/json/O1/{}_{}/{}/{}/'
+        url = url.format(ifo, segment_name, start_time, duration)
+        
+        try:
+            flag_segments = json.loads(urllib.urlopen(url).read())['segments']
+        except:
+            msg = "Unable to find segments in GWOSC, check flag name or times"
+            if source == 'any':
+                return query_flag(ifo, segment_name, start_time, end_time,
+                                  source=='dqsegdb')
+            else:
+                raise ValueError(msg)
+    elif source == 'dqsegdb':
+        # Let's not hard require dqsegdb to be installed if we never get here.
+        try:
+            from dqsegdb.apicalls import dqsegdbQueryTimes as query
+        except ImportError:
+            raise ValueError("Could not query flag. DQSegdb is not installed" 
+                             "to additionally try there")
+
+        # The veto definer will allow the use of MACRO names
+        # These directly correspond the name defined in the veto definer file.
+        if veto_definer is not None:
+            veto_def = parse_veto_definer(veto_definer)
+
+        # We treat the veto definer name as if it were its own flag and
+        # a process the flags in the veto definer
+        flag_segments = segmentlist([])
+        if veto_definer is not None and segment_name in veto_def[ifo]:
```

Comment:
If I'm right you're going to be querying the entire veto-definer file multiple times. This is a slow step and should only be run once.